cargo-shuttle/src/lib.rs

```diff
@@ -604,6 +604,29 @@ impl Shuttle {
                 runtime_path
             } else {
                 trace!(path = ?executable_path, "using alpha runtime");
+                if let Err(err) = check_version(&executable_path) {
+                    warn!("{}", err);
+                    if let Some(mismatch) = err.downcast_ref::<VersionMismatchError>() {
+                        println!("Warning: {}.", mismatch);
+                        if mismatch.shuttle_runtime > mismatch.cargo_shuttle {
+                            // The runtime is newer than cargo-shuttle so we
+                            // should help the user to update cargo-shuttle.
+                            println!(
+                                "[HINT]: You should update cargo-shuttle. \
+                                If cargo-shuttle was installed using cargo, \
+                                you can get the latest version by running \
+                                `cargo install cargo-shuttle`."
+                            );
+                        } else {
+                            println!(
+                                "[HINT]: You should update shuttle-runtime. \
+                                Change its version to {} in this project's \
+                                Cargo.toml to use a compatible version.",
```

Comment:
Nit: it's not necessarily incompatible, so we could rephrase it to this (or something like this).
```suggestion
                                "[HINT]: A newer version of shuttle-runtime is available. \
                                Change its version to {} in this project's Cargo.toml to update it.",
```